node/pkg/algorand/watcher.go

```diff
@@ -2,29 +2,250 @@ package algorand
+				EmitterChain:     vaa.ChainIDAlgorand,
+				EmitterAddress:   a,
+				Payload:          at.ApplicationArgs[1],
+				ConsistencyLevel: 32,   // What SHOULD this be?
+			}
+			
+			algorandMessagesConfirmed.Inc()
+
+			logger.Info("message observed",
+				zap.Time("timestamp", observation.Timestamp),
+				zap.Uint32("nonce", observation.Nonce),
+				zap.Uint64("sequence", observation.Sequence),
+				zap.Stringer("emitter_chain", observation.EmitterChain),
+				zap.Stringer("emitter_address", observation.EmitterAddress),
+				zap.Binary("payload", observation.Payload),
+				zap.Uint8("consistency_level", observation.ConsistencyLevel),
+			)
+
+			e.msgChan <- observation
+		}
+	}
 }
 
 func (e *Watcher) Run(ctx context.Context) error {
-	readiness.SetReady(common.ReadinessAlgorandSyncing)
+	// an odd thing to broadcast... 
+	p2p.DefaultRegistry.SetNetworkStats(vaa.ChainIDAlgorand, &gossipv1.Heartbeat_Network{
+		ContractAddress: fmt.Sprintf("%d", e.appid),
+	})
+
+	logger := supervisor.Logger(ctx)
+	errC := make(chan error)
+
+	logger.Info("Algorand watcher connecting", zap.String("url", e.indexerRPC))
+
+	go func() {
+		timer := time.NewTicker(time.Second * 1)
+		defer timer.Stop()
+
+		indexerClient, err := indexer.MakeClient(e.indexerRPC, e.indexerToken)
+		if err != nil {
+			logger.Info(err.Error())
+			p2p.DefaultRegistry.AddErrorCount(vaa.ChainIDAlgorand, 1)
+			errC <- err
+			return
+		}
+
+		// Parameters
+		var notePrefix = "publishMessage"
 
-	select {}
+		if e.next_round == 0 {
+			// What is the latest round...
+			result, err := indexerClient.SearchForTransactions().Limit(0).Do(context.Background())
+			if err != nil {
+				logger.Info(err.Error())
+				p2p.DefaultRegistry.AddErrorCount(vaa.ChainIDAlgorand, 1)
+				errC <- err
+				return
+			}
+			e.next_round = result.CurrentRound + 1
+			logger.Info("Algorand next_round set to " + fmt.Sprintf("%d", e.next_round))
+		}
+
+		for {
+			select {
+			case <-ctx.Done():
+				return
+			case r := <-e.obsvReqC:
+				if vaa.ChainID(r.ChainId) != vaa.ChainIDAlgorand {
+					panic("invalid chain ID")
+				}
+
+				logger.Info("Received obsv request: " + hex.EncodeToString(r.TxHash) + " -> " + base32.StdEncoding.WithPadding(base32.NoPadding).EncodeToString(r.TxHash))
+
+				result, err := indexerClient.SearchForTransactions().TXID(base32.StdEncoding.WithPadding(base32.NoPadding).EncodeToString(r.TxHash)).Do(context.Background())
+				if err != nil {
+					logger.Info(err.Error())
+					p2p.DefaultRegistry.AddErrorCount(vaa.ChainIDAlgorand, 1)
+					errC <- err
+					return
+				}
+				for i := 0; i < len(result.Transactions); i++ {
+					var t = result.Transactions[i]
+					lookAtTxn(e, t, logger)
+				}
+
+			case <-timer.C:
+				var nextToken = ""
+				for {
+					result, err := indexerClient.SearchForTransactions().NotePrefix([]byte(notePrefix)).MinRound(e.next_round).NextToken(nextToken).Do(context.Background())
+					if err != nil {
+						logger.Info(err.Error())
```

Comment:
```suggestion
						logger.Error("problem searching for transactions", zap.Error(err))
```

Perhaps something closer to this?

Reply:
edit made